homeassistant/components/sensor/monzo.py

```diff
@@ -0,0 +1,364 @@
+"""
+Support for the Monzo API.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.monzo/
+"""
+import os
+import logging
+import datetime
+import time
+
+import voluptuous as vol
+
+from homeassistant.core import callback
+from homeassistant.components.http import HomeAssistantView
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import ATTR_ATTRIBUTION
+from homeassistant.const import ATTR_ID
+from homeassistant.const import CONF_ID
+from homeassistant.helpers.entity import Entity
+import homeassistant.helpers.config_validation as cv
+from homeassistant.util.json import load_json, save_json
+
+
+REQUIREMENTS = ['monzotomtest==0.6.1']
+
+_CONFIGURING = {}
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_ACCESS_TOKEN = 'access_token'
+ATTR_REFRESH_TOKEN = 'refresh_token'
+ATTR_CLIENT_ID = 'client_id'
+ATTR_CLIENT_SECRET = 'client_secret'
+ATTR_LAST_SAVED_AT = 'last_saved_at'
+
+CONF_MONITORED_RESOURCES = 'monitored_resources'
+CONF_ATTRIBUTION = 'Data provided by Monzo.com'
+
+DEPENDENCIES = ['http']
+
+MONZO_AUTH_CALLBACK_PATH = '/api/monzo/callback'
+MONZO_AUTH_START = '/api/monzo'
+MONZO_CONFIG_FILE = 'monzo.conf'
+MONZO_DEFAULT_RESOURCES = ['balance']
+
+SCAN_INTERVAL = datetime.timedelta(minutes=30)
+
+DEFAULT_CONFIG = {
+    'client_id': 'CLIENT_ID_HERE',
+    'client_secret': 'CLIENT_SECRET_HERE'
+}
+
+MONZO_RESOURCES_LIST = {
+    'balance': ['Account Balance', 'GBP', 'cash'],
+    'dailyspend': ['Spent Today', 'GBP', 'cash'],
+    'pots': ['Balance', 'GBP', 'cash']
+}
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_MONITORED_RESOURCES, default=MONZO_DEFAULT_RESOURCES):
+        vol.All(cv.ensure_list, [vol.In(MONZO_RESOURCES_LIST)]),
+    vol.Optional(CONF_ID, default=""): cv.string
+})
+
+
+def request_app_setup(hass, config, add_entities, config_path,
+                      discovery_info=None):
+    """Assist user with configuring the Monzo dev application."""
+    configurator = hass.components.configurator
```

Comment:
The configurator is deprecated. We want to use config entries now.

https://developers.home-assistant.io/docs/en/config_entries_index.html

Reply:
Ok, I'll have a look at that and update. Thanks.